### PR TITLE
Changed HostGroup decode to match Host

### PR DESCRIPTION
**Pull request by lhw**
Fixes issue with parameter decoding in HostGroups previously hidden by other issues.

### CODE REVIEW

File: foreman/api/hostgroup.go [hunk 1]
```diff
@@ -73,121 +73,17 @@ type ForemanHostgroup struct {
...
-	jsonDecErr = json.Unmarshal(b, &fhMap)
-	if jsonDecErr != nil {
-		return jsonDecErr
-	}
-
-	var ok bool
-	if fh.Title, ok = fhMap["title"].(string); !ok {
-		fh.Title = ""
-	}
-	if fh.RootPassword, ok = fhMap["root_password"].(string); !ok {
```
Comment (MrFreezeex) [inline foreman/api/hostgroup.go:163]: Hmm apparently there is `root_pass` and `root_password` in the code :(.

Comment (lhw) [reply foreman/api/hostgroup.go:163]: Okay thats not the great from a comprehension perspective. But I think it should still be technically okay. The resource mapping seems to be correct.

Comment (MrFreezeex) [reply foreman/api/hostgroup.go:163]: No but I mean in decoding foreman use `root_password` and in encode it uses `root_pass`... So I guess that you would need to reflect that in `foremanHostGroupDecode` (except if this code is buggy somehow)

Comment (lhw) [reply foreman/api/hostgroup.go:163]: The mapping from API (`root_pass`) via `ForemanHostgroup` to api.ForemanHostgroup.RootPassword should be fine. The only reference to `root_password` is in the resource. 

The way back via resource_foreman_hostgroup.go:251 should be fine too.
```go
	if attr, ok = d.GetOk("root_password"); ok {
		hostgroup.RootPassword = attr.(string)
	}
```
Am I missing something?

Comment (MrFreezeex) [reply foreman/api/hostgroup.go:163]: Well the line that I just commented is using `root_password` as a key so it's not just in the resource apparently... Although I checked the API doc of foreman and I don't see `root_password` anywhere which is weird... Maybe this code was buggy somehow?

Comment (lhw) [reply foreman/api/hostgroup.go:163]: You are correct. But that line was not really correct anyway. The root_pass key is never returned from the API. Its a write-only field in the API.

Comment (MrFreezeex) [reply foreman/api/hostgroup.go:163]: Ok fine that's a good thing then :). I cross checked quickly the other key and I don't think there are others inconsistencies.

**Pull request merged**